prboom2/src/gl_main.c

```diff
@@ -2253,7 +2272,23 @@ static void gld_DrawSprite(GLSprite *sprite)
     {
       glGetIntegerv(GL_BLEND_SRC, &blend_src);
       glGetIntegerv(GL_BLEND_DST, &blend_dst);
-      glBlendFunc(GL_DST_COLOR, GL_ONE_MINUS_SRC_ALPHA);
+      switch (gl_thingspritefuzzmode)
+      {
```

Comment:
Do you think we could achieve the same with function pointers instead of checking the same condition for each sprite during each frame?

Reply:
I considered it, but did not do it for the reason that, really the entire thing should be "functionized" at that point:

```c
gl_SaveBlendSettings();
gl_SetSpriteFuzzMode(gl_{whatever}spritefuzzmode);
/* ... */
gl_RestoreBlendSettings();
```

And rather than do such surgery I went with the duplication.

If it is preferred another way, I am of course open to it. I can simply move the blend function calls to a separate function instead. But I think at some point it should get cleaned up, because there are other transparency modes that are not in use that will require more save/restore settings, and if those get used later on, it will require untangling what is done here.

Reply:
FYI: The latest branch has the blend functions in a call. I think for now this should suffice, though note that in the future if we add more blending options (say with new shaders or such), this would need to be revisited.